From 35ebfed93dc13874175734e2901dd76d8714e697 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 12 Sep 2012 13:10:30 -0300
Subject: [RHEL6 qemu-kvm PATCH 11/14] usb-redir: Return babble when getting
 more bulk data then requested

RH-Author: Hans de Goede <hdegoede@redhat.com>
Message-id: <1347455431-11260-12-git-send-email-hdegoede@redhat.com>
Patchwork-id: 41854
O-Subject: [RHEL-6.4 qemu-kvm PATCH 11/12] usb-redir: Return babble when getting more bulk data then requested
Bugzilla:
RH-Acked-by: Alon Levy <alevy@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Christophe Fergeau <cfergeau@redhat.com>
Bugzilla-related: 805172

Babble is the appropriate error in this case (rather then signalling a stall).

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Upstream-commit: 2979a36183a3902cd75665e7c6bbc8668668fd17
Conflicts: usb-redir.c
---
 usb-redir.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 usb-redir.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usb-redir.c b/usb-redir.c
index fe7254d..795177d 100644
--- a/usb-redir.c
+++ b/usb-redir.c
@@ -1400,9 +1400,9 @@ static void usbredir_bulk_packet(void *priv, uint32_t id,
             if (data_len <= aurb->packet->len) {
                 memcpy(aurb->packet->data, data, data_len);
             } else {
-                ERROR("bulk buffer too small (%d > %d)\n", data_len,
-                      aurb->packet->len);
-                len = USB_RET_STALL;
+                ERROR("bulk got more data then requested (%d > %d)\n",
+                      data_len, aurb->packet->len);
+                len = USB_RET_BABBLE;
             }
         }
         aurb->packet->len = len;
-- 
1.7.11.4

